File: frontend/src/lib/paymentMethods.ts
// Labels en español para los valores de payment_method. El valor
// guardado en la DB es un slug estable en inglés (no se traduce ahí
// para no romper la lógica de caja/cuenta corriente en el backend que
// compara contra "cash"/"account" literal) -- esto solo mapea a texto
// legible para mostrar en la UI. Espeja backend/app/shared/payment_methods.py.

export const PAYMENT_METHODS = ["cash", "card_debit", "card_credit", "transfer", "account"] as const;

export type PaymentMethod = (typeof PAYMENT_METHODS)[number];

const PAYMENT_METHOD_LABELS: Record<string, string> = {
  cash: "Efectivo",
  card_debit: "Tarjeta de débito",
  card_credit: "Tarjeta de crédito",
  transfer: "Transferencia",
  account: "Cuenta corriente",
};

export function paymentMethodLabel(method: string): string {
  return PAYMENT_METHOD_LABELS[method] ?? method;
}

export function isCardPayment(method: string): boolean {
  return method === "card_debit" || method === "card_credit";
}

File: backend/app/shared/payment_methods.py
"""Labels en español para los valores de payment_method usados en Sale
y CustomerPayment. El valor guardado en la DB es un slug estable en
inglés (no se traduce ahí para no romper los chequeos existentes como
CASH_PAYMENT_METHOD/ACCOUNT_PAYMENT_METHOD en treasury/services.py) --
esto solo mapea a texto legible para mostrar/imprimir.
"""

PAYMENT_METHOD_LABELS: dict[str, str] = {
    "cash": "Efectivo",
    "card_debit": "Tarjeta de débito",
    "card_credit": "Tarjeta de crédito",
    "transfer": "Transferencia",
    "account": "Cuenta corriente",
}


def payment_method_label(payment_method: str) -> str:
    return PAYMENT_METHOD_LABELS.get(payment_method, payment_method)


def is_card_payment(payment_method: str) -> bool:
    return payment_method in ("card_debit", "card_credit")
